lib/iris/tests/test_image_json.py

```diff
@@ -34,32 +34,47 @@
-
-        iuri = ('https://api.github.com/repos/scitools/test-iris-imagehash/'
-                'contents/images')
         # https://developer.github.com/v3/#user-agent-required
-        headers = {'User-Agent': 'SciTools'}
-        r = requests.get(iuri, headers=headers)
-        if r.status_code != 200:
-            raise ValueError('Github API get failed: {}'.format(iuri))
-        rj = r.json()
-        prefix = 'https://scitools.github.io/test-iris-imagehash/images/'
+        headers = {'User-Agent': 'scitools-bot'}
+        rate_limit_uri = 'https://api.github.com/rate_limit'
+        rl = requests.get(rate_limit_uri, headers=headers)
+        some_left = False
+        if rl.status_code == 200:
+            rates = rl.json()
+            remaining = rates.get('rate', {})
+            remaining = remaining.get('remaining')
+            if isinstance(remaining, int):
+                if remaining > 3:
+                    some_left = True
+        # Only run this test if there are IP based rate limited calls left.
+        if some_left:
+            iuri = ('https://api.github.com/repos/scitools/'
+                    'test-iris-imagehash/contents/images')
+            r = requests.get(iuri, headers=headers)
+            if r.status_code != 200:
+                raise ValueError('Github API get failed: {}'.format(iuri,
+                                                                    r.text))
+            rj = r.json()
+            prefix = 'https://scitools.github.io/test-iris-imagehash/images/'
 
-        known_image_uris = set([prefix + rji['name'] for rji in rj])
+            known_image_uris = set([prefix + rji['name'] for rji in rj])
 
-        repo_fname = os.path.join(os.path.dirname(__file__), 'results',
-                                  'imagerepo.json')
-        with open(repo_fname, 'rb') as fi:
-            repo = json.load(codecs.getreader('utf-8')(fi))
-        uris = set(itertools.chain.from_iterable(six.itervalues(repo)))
+            repo_fname = os.path.join(os.path.dirname(__file__), 'results',
+                                      'imagerepo.json')
+            with open(repo_fname, 'rb') as fi:
+                repo = json.load(codecs.getreader('utf-8')(fi))
+            uris = set(itertools.chain.from_iterable(six.itervalues(repo)))
 
-        amsg = ('Images are referenced in imagerepo.json but not published in '
-                'https://scitools.github.io/test-iris-imagehash/images:\n{}')
-        amsg = amsg.format('\n'.join(list(uris.difference(known_image_uris))))
+            amsg = ('Images are referenced in imagerepo.json but not published'
+                    ' in https://scitools.github.io/test-iris-imagehash/'
+                    'images:\n{}')
+            diffs = list(uris.difference(known_image_uris))
+            amsg = amsg.format('\n'.join(diffs))
 
-        self.assertTrue(uris.issubset(known_image_uris), msg=amsg)
+            self.assertTrue(uris.issubset(known_image_uris), msg=amsg)
```

Comment:
> At least that would be captured in the Travis CI log ...

I did think about this, but the travis log is very busy, i wasn't sure it would be easy to find

i can add it in easily if you think it adds value